### ISSUE TITLE
From a shared folder the workspace (readme.md) is not displayed.

### ISSUE THREAD

**Issue by the-matzup** [nextcloud/server#20028]
### Problem
The new "workspace" (readme.md) will not be displayed for all users who only receive a public link and are not logged in.

### Steps to reproduce
1. Create folder with subfolders.
2. Fill the upper workspace for the folders with text.
3. Create a public link for the folder.
4. After the public link has been opened: You will see the file "Readme.md" but not automatically the content above the files/folders as it should be.

Note: The "rich workspace" in the settings is activated!
Note2: Tested with different browsers (Safari, Firefox, Chrome) on different systems (iOS, Mac, Win, Android) 

### Expected behaviour
Displayed the readme.md content above the files/folders as it should be – even if you are not logged in.

### Actual behaviour
The content of the readme.md is not displayed when you get a public link.

### Server configuration

**Nextcloud version:**
18.0.2

**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh install

http://example.com/index.php/settings/integrity/failed :
```
No errors have been found.
```

**Comment by kesselb** [comment]
I think this works exactly as designed. Workspaces are not a etherpad replacement. 

cc @juliushaertl @jancborchardt

**Comment by juliushaertl** [comment]
No, it was supposed to work exactly like that, so this seems like a recent bug. I'll have a look at that.
